### DISCUSSION TITLE
change primary key

### DISCUSSION THREAD

**Discussion by pandherjs** [serverpod/serverpod#1986] [Q&A] [answered]
Hey guys!
I researched to find the answer for this question however i couldn't.
Question-> How can I keep id series in table but also add userID which will be generated auto and will be int (maybe string if Serverpod allow i want to use username)

Here are my models and example:
user.spy.yaml
```
class: User
table: user
fields:
    userId: int 
    username: String
indexes:
    user_idx:
        fields: username
        unique: true
```

feed.spy.yaml

```
class: Feed
table: feed
fields:
  title: String
  userId: User?, relation
  timeStamp: DateTime
```

I want to use UserId instead of id (series) to make the relation and allow my endpoints to find or insert using userId instead of id.

Basically, I dont want to send series id to client and keep it unique and random.

Thanks!!

**Comment by SandPod** [comment, marked as answer]
Hey @pandherjs 👋 

Unfortunately, Serverpod doesn't support using an arbitrary field as a foreign key for the relation yet.
This is the issue that tracks the addition of this feature: https://github.com/serverpod/serverpod/issues/1147

The object relation interface at the moment requires the relation to be based on the automatically generated id of the table.
